src/cluster/src/check/mod.rs

```diff
@@ -374,30 +593,31 @@ impl ClusterChecker {
     /// # use fluvio_cluster::ClusterChecker;
     /// let checker: ClusterChecker = ClusterChecker::empty();
     /// ```
+    ///
+    /// [`with_check`]: ClusterChecker::with_check
     pub fn empty() -> Self {
         ClusterChecker { checks: vec![] }
     }
 
     /// Adds a check to this `ClusterChecker`
-    pub fn with_check(mut self, check: Box<dyn ClusterCheck>) -> Self {
-        self.checks.push(check);
+    pub fn with_check<C: ClusterCheck, B: Into<Box<C>>>(mut self, check: B) -> Self {
+        self.checks.push(check.into());
         self
     }
 
     /// Adds all preflight checks to this checker.
     ///
     /// Note that no checks are run until one of the `run` methods are invoked.
     ///
-    /// - [`run_wait`]
-    /// - [`run_wait_and_fix`]
-    /// - [`run_with_progress`]
-    /// - [`run_and_fix_with_progress`]
+    /// - [`run_wait`](ClusterChecker::run_wait)
+    /// - [`run_wait_and_fix`](ClusterChecker::run_wait_and_fix)
+    /// - [`run_with_progress`](ClusterChecker::run_with_progress)
+    /// - [`run_and_fix_with_progress`](ClusterChecker::run_and_fix_with_progress)
     pub fn with_preflight_checks(mut self) -> Self {
         let checks: Vec<Box<(dyn ClusterCheck)>> = vec![
             Box::new(LoadableConfig),
             Box::new(K8Version),
             Box::new(HelmVersion),
-            Box::new(SysChart),
```

Comment:
why remove sys chart check?

Reply:
I only removed it from this helper method because it now requires a `SysConfig` object with the namespace and other info required to install the sys chart, and we do not have that information available here. If you look at where this is actually called (from k8 installer and local installer), the SysChartCheck is now manually added to the checker using a config that is derived from the installer's config itself.

https://github.com/infinyon/fluvio/pull/721/files#diff-acfbf9dfd3f6cc9ca6cba75c1b476d4144794784b4c93ef7c8e8d716c5a65591R333

This means that the SysConfig is now derived from the K8Install config or the LocalInstall config, so it will inherit certain options rather than them being hardcoded (e.g. the namespace).